airflow/www/views.py

```diff
@@ -836,6 +839,21 @@ def log(self, session=None):
         dttm = pendulum.parse(execution_date)
         form = DateTimeForm(data={'execution_date': dttm})
         dag = dagbag.get_dag(dag_id)
+        metadata = {}
+        tailing_required = False
+        tail_lines_list = [100, 200, 500, 1000]
+        num_lines = 100
+        if conf.has_option('webserver', 'tail_lines_list'):
+            tail_lines_list = conf.get('webserver', 'tail_lines_list')
+            tail_lines_list = list(map(int, tail_lines_list.split(',')))
+        if conf.has_option('webserver', 'tail_logs'):
+            tailing_required = conf.getboolean('webserver', 'tail_logs')
+            if tailing_required and conf.has_option('webserver', 'num_lines'):
+                num_lines = conf.getint('webserver', 'num_lines')
+            metadata["num_lines"] = num_lines
+        if num_lines in tail_lines_list:
+            tail_lines_list.remove(num_lines)
```

Comment:
What is this conditional for? I can't work it out

Reply:
That's to avoid duplicates. lets say user have added in config that dropdown should contain entries 100,200,500,1000 and user picked 100 as default number of lines to tail when user lands on get_logs page. Without this condition airflow ending showing 2 entries for 100 in dropdown. This will ensure no duplicates in such case